homeassistant/components/sensor/mold_indicator.py

```diff
@@ -0,0 +1,255 @@
+MAGNUS_K3 = 243.12
+
+ATTR_DEWPOINT = "Dewpoint"
+ATTR_CRITICAL_TEMP = "Est. Crit. Temp"
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Setup MoldIndicator sensor."""
+    name = config.get('name', DEFAULT_NAME)
+    indoor_temp_sensor = config.get(CONF_INDOOR_TEMP)
+    outdoor_temp_sensor = config.get(CONF_OUTDOOR_TEMP)
+    indoor_humidity_sensor = config.get(CONF_INDOOR_HUMIDITY)
+    calib_value = util.convert(config.get(CONF_CALIBRATION_FACTOR),
+                               float, None)
+
+    if None in (indoor_temp_sensor,
+                outdoor_temp_sensor, indoor_humidity_sensor):
+        _LOGGER.error('Missing required key %s, %s or %s',
+                      CONF_INDOOR_TEMP, CONF_OUTDOOR_TEMP,
+                      CONF_INDOOR_HUMIDITY)
+        return False
+
+    add_devices_callback([MoldIndicator(
+        hass, name, indoor_temp_sensor,
+        outdoor_temp_sensor, indoor_humidity_sensor,
+        calib_value)])
+
+
+# pylint: disable=too-many-instance-attributes
+class MoldIndicator(Entity):
+    """Represents a MoldIndication sensor."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, hass, name, indoor_temp_sensor, outdoor_temp_sensor,
+                 indoor_humidity_sensor, calib_value):
+        """Initialize the sensor."""
+        self._state = "-"
+        self._hass = hass
+        self._name = name
+        self._indoor_temp_sensor = indoor_temp_sensor
+        self._indoor_humidity_sensor = indoor_humidity_sensor
+        self._outdoor_temp_sensor = outdoor_temp_sensor
+        self._calib_value = calib_value
+        self._is_metric = (hass.config.temperature_unit == TEMP_CELCIUS)
+
+        self._dewpoint = None
+        self._indoor_temp = None
+        self._outdoor_temp = None
+        self._indoor_hum = None
+        self._crit_temp = None
+
+        track_state_change(hass, indoor_temp_sensor, self._sensor_changed)
+        track_state_change(hass, outdoor_temp_sensor, self._sensor_changed)
+        track_state_change(hass, indoor_humidity_sensor, self._sensor_changed)
+
+        # Read initial state
+        self._sensor_changed(indoor_temp_sensor,
+                             None, hass.states.get(indoor_temp_sensor))
+        self._sensor_changed(outdoor_temp_sensor,
+                             None, hass.states.get(outdoor_temp_sensor))
+        self._sensor_changed(indoor_humidity_sensor,
+                             None, hass.states.get(indoor_humidity_sensor))
+
+    @staticmethod
+    def _update_temp_sensor(state):
+        """Parse temperature sensor value."""
+        unit = state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
+        temp = util.convert(state.state, float)
+
+        if temp is None:
+            _LOGGER.error('Unable to parse sensor temperature: %s',
+                          state.state)
+            return None
+
+        # convert to celsius if necessary
```

Comment:
Home Assistant will take care that all published states to the state machine are converted to the unit of the user.